### PR TITLE
CUMULUS-1641: Pop SQS trigger messages off stack after successful execution

### PR DESCRIPTION
**Pull request by jennyhliu**
**Summary:** Summary of changes

Addresses [CUMULUS-1641: Pop SQS trigger messages off stack after successful execution](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1641)

## PR Checklist

- [x ] Update CHANGELOG
- [x ] Unit tests
- [x ] Adhoc testing
- [x ] Integration tests

### CODE REVIEW

File: packages/api/lambdas/sqs-message-consumer.js [hunk 1]
```diff
@@ -42,8 +48,35 @@ async function processQueues(event, dispatchFn) {
  *
  */
 function dispatch(message) {
-  const input = Object.assign({}, message.Body);
-  return rulesHelpers.queueMessageForRule(this.rule, input);
+  const queueUrl = this.rule.rule.value;
+  const messageReceiveCount = parseInt(message.Attributes.ApproximateReceiveCount, 10);
+
+  if (get(this.rule, 'meta.retries', 3) < messageReceiveCount - 1) {
+    log.debug(`message ${message.MessageId} from queue ${queueUrl} has been processed ${messageReceiveCount - 1} times, no more retries`);
```
Comment (laurenfrederick) [inline packages/api/lambdas/sqs-message-consumer.js]: What is this doing here? We're saying no more retries but putting it back on the queue? Should this go to a DLQ?

File: packages/api/lambdas/sqs-message-remover.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  return true;
+}
+
+/**
+ * Update SQS queue when workflow of the message is completed
+ *
+ * @param {Object} event - Cloudwatch event
+ */
+async function updateSqsQueue(event) {
+  if (!isSqsQueueUpdateNeeded(event)) return 'Not a valid event for updating SQS queue';
```
Comment (laurenfrederick) [inline packages/api/lambdas/sqs-message-remover.js]: This is returning a string, but we expect this function to return an object. I would either log  that message and return `Promise.resolve()` or throw an error.

File: packages/api/lambdas/sqs-message-remover.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+ * Lambda handler for sqs-message-remover Lambda
+ *
+ * This Lambda function works together with sqs-message-consumer. sqs-message-consumer lambda
+ * consumes message from SQS queue, and sqs-message-remover deletes the message from the SQS
+ * queue when the workflow triggered by the message is completed successfully.
+ *
+ * @param {Object} event - Cloudwatch event
+ * @returns {Promise}
+ */
+async function handler(event) {
```
Comment (laurenfrederick) [inline packages/api/lambdas/sqs-message-remover.js]: Is this triggered by listening to the execution CW event notifications?

File: docs/data-cookbooks/setup.md [hunk 4]
```diff
@@ -93,6 +94,13 @@ An example of an SQS rule configuration is [here](https://github.com/nasa/cumulu
 |name|`"L2_HR_PIXC"`|Yes|Name of a collection defined/configured in the Collections dashboard page|
 |version|`"000"`|Yes|Version number of a collection defined/configured in the Collections dashboard page|
 
+### meta-object
+
+|Key  |Value  |Required|Description|
+|:---:|:-----:|:------:|-----------|
+|retries|`3`|No|Number of retries on errors, for sqs-type rule only|
+|visibilityTimeout|`120`|No|VisibilityTimeout in seconds for the inflight messages, for sqs-type rule only|
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply docs/data-cookbooks/setup.md]: The initial `visibilityTimeout` value is set to the `visibilityTimeout` of the queue used to create the rule.
https://github.com/nasa/cumulus/blob/9d840834006e1481458d37ce8e10e862a74680c3/packages/api/models/rules.js#L452

File: packages/api/lambdas/sqs-message-consumer.js (hunk 1 above)
Comment (jennyhliu) [reply packages/api/lambdas/sqs-message-consumer.js]: The message is ignored if the retries limit is reached.  The message will be removed to DLQ by the SQS service based on the RedrivePolicy for the DLQ.

File: packages/api/lambdas/sqs-message-remover.js (hunk 2 above)
Comment (jennyhliu) [reply packages/api/lambdas/sqs-message-remover.js]: I updated the jsdoc.  It should return a `Promise` object.

File: packages/api/lambdas/sqs-message-remover.js (hunk 3 above)
Comment (jennyhliu) [reply packages/api/lambdas/sqs-message-remover.js]: Yes.  I update the doc

File: packages/api/lambdas/sqs-message-consumer.js [hunk 5]
```diff
@@ -22,12 +23,17 @@ async function processQueues(event, dispatchFn) {
...
     const queueUrl = rule.rule.value;
+
+    if (!(await sqsQueueExists(queueUrl))) return Promise.resolve();
+
     const consumer = new Consumer({
       queueUrl: queueUrl,
       messageLimit,
-      timeLimit
+      timeLimit,
+      visibilityTimeout: rule.meta.visibilityTimeout,
```
[the remark this answers is not in the document]
Comment (jennyhliu) [reply packages/api/lambdas/sqs-message-consumer.js]: It's set when the rule is created, default to the visibilityTimeout of the queue if it's not provided.

File: packages/api/lambdas/sqs-message-consumer.js (hunk 1 above)
Comment (laurenfrederick) [reply packages/api/lambdas/sqs-message-consumer.js]: Ok thanks! Tested it out and it works!

**Pull request merged**